graph.cpp

```diff
@@ -40,7 +40,7 @@ Graph::Graph() : started_(false) {
     addGetProperty("started", this, &Graph::isStarted);
 }
 
-bool Graph::addNode(const std::string& name, NodeBase* node) {
+bool Graph::addNode(const std::string& name, std::shared_ptr<NodeBase> node) {
     ScopedLock lock(&mutex_);
     assert(node != 0);
     if (lockedGetNodeByName(name) != 0) {
```

Comment:
[minor] there are a few comparisons with 0, or return 0. Can be replaced to `nullptr`

Reply:
actually shared_ptr has a cast-to-bool operator... so I can simply remove the comparison.

Reply:
ok. I think that == nullptr is more descriptive in this case. both work for me.